src/EventSubscriber/FixedWindowRateLimiterSubscriber.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\EventSubscriber;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Exception\TooManyRequestsHttpException;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\RateLimiter\RateLimiterFactory;
+
+class FixedWindowRateLimiterSubscriber implements EventSubscriberInterface
+{
+    /** @var RateLimiterFactory */
+    private $fixedWindowLimiter;
+
+    /** @var string[] */
+    private $fixedWindowRoutes;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(
+        RateLimiterFactory $fixedWindowLimiter,
+        LoggerInterface $securityLogger,
+        array $fixedWindowRoutes
+    ) {
+        $this->fixedWindowLimiter = $fixedWindowLimiter;
+        $this->fixedWindowRoutes = $fixedWindowRoutes;
+        $this->logger = $securityLogger;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => ['onKernelRequest', 17],
```

Comment:
What is the reason of specifying the priority?

Reply:
The reason is that this event must have more priority than security events (they interrupt the process if the user gets the access token or it is not authenticated for example) and less priority of EventListener\RouterListener because this is the time that the framework sets the attributes on the request (the `_router` attribute).